### PR TITLE
LPS-168326 - Fix the useEffect which returns the Data Source values

### PR DESCRIPTION
**Pull request by Artartur**
[LPS-168326](https://issues.liferay.com/browse/LPS-168326) - The problem was in the `objectDefinition` which was returning underfined as It wasn't going into useEffect which builds `getAllObjectDefinitions` and sets `objectDefinition`, so I repositioned the useEffect and inside it I added the constant and method that returns the Data Source values

### CODE REVIEW

**Comment by Artartur** [comment]
ci:test:relevant

File: modules/apps/notification/notification-web/src/main/resources/META-INF/resources/js/components/Attachments.tsx [hunk 1]
```diff
@@ -76,6 +76,21 @@ export function Attachments({setValues, values}: IProps) {
...
+			const objectDefinitions = items.filter(({system}) => !system);
+
+			setObjectDefinitions(objectDefinitions);
+
+			const currentObjectDefinition = objectDefinitions?.find(
+				(item) => item.id === values.objectDefinitionId
+			);
+
+			setSelectedEntity(currentObjectDefinition);
+		});
```
Comment (danvb112) [inline modules/apps/notification/notification-web/src/main/resources/META-INF/resources/js/components/Attachments.tsx:90]: Hey @Artartur, to follow a pattern we use here we can use async/await notation, as useEffect by default cannot use async notation we can [write the async function inside useEffect](https://devtrium.com/posts/async-functions-useeffect#write-the-asynchronous-function-inside-the-useeffect) and call it at the end, also you can set the states at the end, it makes it easier to read the function, like:


```tsx
useEffect(() => {
  const makeFetch = async () => {
	  const objectDefinitions = await API.getAllObjectDefinitions();
  
	  const currentObjectDefinition = objectDefinitions?.find(
		  (item) => item.id === values.objectDefinitionId
	  );
  
	  setObjectDefinitions(objectDefinitions.filter(({system}) => !system));
	  setSelectedEntity(currentObjectDefinition);
  }
  
  makeFetch();
}, [values.objectDefinitionId]);
```

File: modules/apps/notification/notification-web/src/main/resources/META-INF/resources/js/components/Attachments.tsx [hunk 2]
```diff
@@ -76,6 +76,21 @@ export function Attachments({setValues, values}: IProps) {
...
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, []);
```
Comment (danvb112) [inline modules/apps/notification/notification-web/src/main/resources/META-INF/resources/js/components/Attachments.tsx:92]: Hey @Artartur, you can also remove this `//eslint-disable-next-line react-hooks/exhaustive-deps`, hooks dependencies are very important for React, we use this comment in some cases that suggested dependencies are impacting what we want do, but it's not the case, you can put `values.objectDefinitionId` in the dependencies array

**Review by danvb112 — CHANGES_REQUESTED** [review]
Hey @Artartur, good job on that, I have just two suggestion about, fell free to ask if you have any question :smile:

**Comment by Artartur** [comment]
ci:test:sf

**Review by danvb112 — APPROVED** [review]
LGTM! :tada:

**Comment by danvb112** [comment]
ci:forward

**Comment by Artartur** [comment]
ci:test:relevant

**Comment by Artartur** [comment]
ci:stop

**Comment by danvb112** [comment]
ci:forward

**Pull request closed without merging**